packages/matter.js/src/tlv/TlvSchema.ts

```diff
@@ -40,6 +40,14 @@ export abstract class TlvSchema<T> extends Schema<T, ByteArray> implements TlvSc
     abstract decodeTlvInternalValue(reader: TlvReader, typeLength: TlvTypeLength): T;
 
     abstract encodeTlvInternal(writer: TlvWriter, value: T, tag?: TlvTag): void;
+
+    injectField(value: T, _fieldId: number, _fieldValue: any, _injectChecker: (fieldValue: any | undefined) => boolean): T {
```

Comment:
Putting it into the schema makes sense.

I originally was going to do it as a simple visitor pattern so the types themselves can simply implement `visit(visitor: (node: TlvSchema<any>) => void)` to navigate between nodes.  Then injectField/removeField would be utility functions that use this.

Might make sense to consider that here, feels like it'd simplify quite a bit.  Though I know you have testing and everything done so maybe not worth the effort to change.

Reply:
Would leave for now ... but I see your point